File: src/main/java/com/woomulwoomul/woomulwoomulbackend/domain/base/BasePermanentEntity.java
package com.woomulwoomul.woomulwoomulbackend.domain.base;

import jakarta.persistence.Column;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;

@Getter
@MappedSuperclass
public abstract class BasePermanentEntity extends BaseEntity {

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    protected ServiceStatus serviceStatus;
}

File: src/main/kotlin/com/woomulwoomul/woomulwoomulbackend/domain/base/BasePermanentEntity.kt
package com.woomulwoomul.woomulwoomulbackend.domain.base

import jakarta.persistence.Column
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.MappedSuperclass
import java.time.LocalDateTime

@MappedSuperclass
abstract class BasePermanentEntity(

    @Column(nullable = false, length = 10)
    @Enumerated(EnumType.STRING)
    var status: ServiceStatus = ServiceStatus.ACTIVE,

    createDateTime: LocalDateTime? = null,
    updateDateTime: LocalDateTime? = null,
    ) : BaseEntity(createDateTime, updateDateTime)